x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/shared/get_explain_data.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ml } from '../../../../../services/ml_api_service';
+import { extractErrorMessage } from '../../../../../../../common/util/errors';
+import { DfAnalyticsExplainResponse, FieldSelectionItem } from '../../../../common/analytics';
+import {
+  getJobConfigFromFormState,
+  State,
+} from '../../../analytics_management/hooks/use_create_analytics_form/state';
+
+export interface GetExplainDataReturnType {
+  success: boolean;
+  expectedMemory: string;
+  fieldSelection: FieldSelectionItem[];
+  errorMessage: string;
+}
+
+export const getExplainData = async (formState: State['form']) => {
```

Comment:
Nit: Maybe rename to `fetchExplainData` - I don't think we have a rule for this but maybe we can use `get*` for local/sync getters and `fetch*` for getting remote/async data.

Reply:
That distinction makes sense 👍  Happy to keep naming consistent with that 👌

Reply:
Updated in 8837c39c1d075444f078c6d82434a820b8e2dbe3